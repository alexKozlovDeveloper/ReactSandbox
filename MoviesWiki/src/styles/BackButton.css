.button {
    padding: 8px;
    font-size: 40px;
    background: url(https://static.thenounproject.com/png/95-200.png);
    min-width: 20px;
    min-height: 20px;
    background-size: contain;
    background-repeat: no-repeat;
    margin: 5px;
    cursor: pointer;
}