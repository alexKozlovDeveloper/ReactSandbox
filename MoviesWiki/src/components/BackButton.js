import React, { Component } from "react";

import styles from "../styles/BackButton.css"

function BackButton(props) {
    return  <div className={styles.button}>{props.buttonText}</div>;
}

export default BackButton;