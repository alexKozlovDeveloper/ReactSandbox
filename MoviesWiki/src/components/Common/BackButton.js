import React, { Component } from "react";

import { Link } from 'react-router-dom'

import styles from "../../styles/BackButton.css"

const BackButton = ({ buttonText }) => {
    return <Link to="/" className={styles.button}>{buttonText}</Link>
}

export default BackButton;