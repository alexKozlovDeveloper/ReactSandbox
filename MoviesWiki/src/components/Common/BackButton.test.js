import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import BackButton from "./BackButton";

configure({adapter: new Adapter()});

test('BackButton rendering test', () => {

    var buttonText = "someButtonText";

    const output = shallow(<BackButton buttonText={buttonText} />);
    expect(shallowToJson(output)).toMatchSnapshot();
 });

