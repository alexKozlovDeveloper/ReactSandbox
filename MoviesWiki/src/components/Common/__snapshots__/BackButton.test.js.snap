// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton rendering test 1`] = `
<div
  className="button"
>
  someButtonText
</div>
`;
